core/src/raw/adapters/kv/backend.rs

```diff
@@ -182,19 +185,60 @@ impl<S: Adapter> Access for Backend<S> {
     fn blocking_list(&self, path: &str, args: OpList) -> Result<(RpList, Self::BlockingLister)> {
         let p = build_abs_path(&self.root, path);
         let res = self.kv.blocking_scan(&p)?;
-        let lister = KvLister::new(&self.root, res);
+        let lister = BlockingKvLister::new(&self.root, res);
         let lister = HierarchyLister::new(lister, path, args.recursive());
 
         Ok((RpList::default(), lister))
     }
 }
 
-pub struct KvLister {
+pub struct KvLister<Iter> {
+    root: String,
+    inner: Mutex<Iter>,
```

Comment:
We don't need `Mutex` here, it must be `Sync` as long as it's `Send`. Since we can only call `next` over it.

Reply:
Done.